Library/Homebrew/dev-cmd/livecheck.rb

```diff
@@ -0,0 +1,459 @@
+# frozen_string_literal: true
+
+require "cli/parser"
+require "livecheck_strategy"
+
+module Homebrew
+  module_function
+
+  WATCHLIST_PATH = (ENV["HOMEBREW_LIVECHECK_WATCHLIST"] || Pathname.new(Dir.home)/".brew_livecheck_watchlist").freeze
+  private_constant :WATCHLIST_PATH
+
+  GITHUB_SPECIAL_CASES = %w[
+    api.github.com
+    /latest
+    mednafen
+    camlp5
+    kotlin
+    osrm-backend
+    prometheus
+    pyenv-virtualenv
+    sysdig
+    shairport-sync
+    yuicompressor
+  ].freeze
+  private_constant :GITHUB_SPECIAL_CASES
+
+  UNSTABLE_VERSION_KEYWORDS = %w[
+    alpha
+    beta
+    bpo
+    dev
+    experimental
+    prerelease
+    preview
+    rc
+  ].freeze
+  private_constant :UNSTABLE_VERSION_KEYWORDS
+
+  NO_VERSIONS_MSG = "Unable to get versions"
+  HEAD_ONLY_MSG = "HEAD only formula must be installed to be livecheckable"
+  private_constant :NO_VERSIONS_MSG, :HEAD_ONLY_MSG
+
+  @livecheck_strategy_names = {}
+
+  def livecheck_args
+    Homebrew::CLI::Parser.new do
+      usage_banner <<~EOS
+        `livecheck` [<formulae>]
+
+        Check for newer versions of formulae from upstream.
+
+        If no formula argument is passed, the list of formulae to check is taken from `HOMEBREW_LIVECHECK_WATCHLIST`
+        or `~/.brew_livecheck_watchlist`.
+      EOS
+      switch :verbose
+      switch :quiet
+      switch :debug
+      switch "--full-name",
+             description: "Print formulae with fully-qualified names."
+      flag   "--tap=",
+             description: "Check the formulae within the given tap, specified as <user>`/`<repo>."
+      switch "--installed",
+             description: "Check formulae that are currently installed."
+      switch "--json",
+             description: "Output informations in JSON format."
+      switch "--all",
+             description: "Check all available formulae."
+      switch "--newer-only",
+             description: "Show the latest version only if it's newer than the formula."
+      conflicts "--debug", "--json"
+      conflicts "--tap=", "--all", "--installed"
+    end
+  end
+
+  def livecheck
```

Comment:
Would be good to have this function be as minimal as possible and essentially just parse/setup arguments and pass them to a method in the `Homebrew::Livecheck` module to actually do the main livecheck logic.